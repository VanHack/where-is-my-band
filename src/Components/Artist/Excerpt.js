import styled from "styled-components";

const Excerpt = styled.div`
  margin-top: 15px;
  font-size: 16px;
  width: 100%;
  margin: auto;
`;

export default Excerpt;
