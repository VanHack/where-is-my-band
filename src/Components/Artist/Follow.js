import styled from "styled-components";

const Follow = styled.a`
  text-align: center;
  color: #4a90e2;
  display: block;
  font-size: 16px;
  width: 100%;
  margin: auto;
  margin-top: 15px;
`;

export default Follow;
