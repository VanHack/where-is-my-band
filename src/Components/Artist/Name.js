import styled from "styled-components";

const Name = styled.h1`
  color: ${props => props.theme.primary};
  font-size: 40px;
  font-weight: bold;
  width: 100%;
  margin: auto;
  margin-top: 15px;
`;

export default Name;
