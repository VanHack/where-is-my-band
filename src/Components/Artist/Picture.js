import styled from "styled-components";

const Picture = styled.div`
  width: 100%;
  height: 300px;
  background-image: url(${props => props.src});
  background-size: cover;
  background-position: center;
  border: 0px;
  margin: auto;
  display: block;
`;

export default Picture;
