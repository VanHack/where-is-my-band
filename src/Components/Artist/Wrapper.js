import styled from "styled-components";

const Wrapper = styled.div`
  display: flex;
  width: 100%;
  padding: 15px;
  flex-direction: column;

  @media (min-width: 600px) {
    width: 50%;
  }
  @media (min-width: 1200px) {
    width: 33.3%;
  }
`;

export default Wrapper;
