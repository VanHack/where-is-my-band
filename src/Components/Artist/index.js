import React from "react";
import styled from "styled-components";

const Wrapper = styled.div`
  display: flex;
  width: 100%;
  padding: 15px;
  flex-direction: column;

  @media (min-width: 600px) {
    width: 50%;
  }
  @media (min-width: 1200px) {
    width: 33.3%;
  }
`;

const Picture = styled.img`
  width: 100%;
  height: auto;
  margin: auto;
  display: block;
`;

const Name = styled.h1`
  color: ${props => props.theme.primary};
  font-size: 40px;
  font-weight: bold;
  width: 100%;
  margin: auto;
  margin-top: 15px;
`;

const Excerpt = styled.div`
  margin-top: 15px;
  font-size: 16px;
  width: 100%;
  margin: auto;
`;

const Follow = styled.a`
  text-align: center;
  color: #4a90e2;
  display: block;
  font-size: 16px;
  width: 100%;
  margin: auto;
  margin-top: 15px;
`;

const Artist = ({ artist }) => (
  <Wrapper>
    <Picture src={artist.image_url} />
    <Name>{artist.name}</Name>
    <Excerpt>{artist.excerpt}</Excerpt>
    <Follow href={artist.facebook_page_url} target="_blank">
      Follow on Facebook
    </Follow>
  </Wrapper>
);

export default Artist;
