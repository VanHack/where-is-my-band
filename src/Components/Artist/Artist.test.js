import React from 'react';
import Artist from './index';
import renderer from 'react-test-renderer';

test('renders an event completely', () => {
  const rockStar = {
    image_url: 'http://example.com/some-image.jpg',
    name: 'Rock Star!',
    excerpt: 'Rock Star band description',
    facebook_page_url: 'http://facebook.com/rockstar',
  }


  const component = renderer.create(
    <Artist artist={rockStar}/>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
