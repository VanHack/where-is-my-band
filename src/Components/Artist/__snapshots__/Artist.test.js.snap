// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an event completely 1`] = `
<div
  className="sc-bdVaJa lmDREB"
>
  <img
    className="sc-htpNat fIYWjR"
    src="http://example.com/some-image.jpg"
  />
  <h1
    className="sc-bxivhb hGpxWJ"
  >
    Rock Star!
  </h1>
  <div
    className="sc-bwzfXH hAsZIe"
  >
    Rock Star band description
  </div>
  <a
    className="sc-ifAKCX goMAVo"
    href="http://facebook.com/rockstar"
    target="_blank"
  >
    Follow on Facebook
  </a>
</div>
`;
